st2common/st2common/util/config_loader.py

```diff
@@ -98,6 +100,20 @@ def _get_values_for_config(self, config_schema_db, config_db):
         config = self._assign_default_values(schema=schema_values, config=config)
         return config
 
+    @staticmethod
+    def _get_object_property_schema(object_schema, init_additional_properties=None):
+        additional_properties = object_schema.get("additionalProperties", {})
+        if additional_properties and isinstance(additional_properties, dict):
+            property_schema = defaultdict(lambda: additional_properties)
+        else:
+            property_schema = {}
+        if init_additional_properties:
+            # ensure that these keys are present in the object (vs just defaultdict)
+            for key in init_additional_properties:
+                property_schema.__missing__(key)
```

Comment:
Could we instead use regular for loop here + ``property_schema[key] = additional_properties``? Or we would then also need to use copy.copy?

Just thinking out loud since that code is more standard and may be easier to understand than relying on internal ``__missing__`` method.

Reply:
And I guess if we just always init all the keys, we would not even need to use ``defaultdict``? Or am I missing some scenario?

Reply:
Sure. I added this section when defaultdict wasn't enough for applying the default values.

Reply:
That ended up much cleaner. Thanks!